src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/jobs/TimeOutLicencesService.kt

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.jobs
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AuditEvent
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.CrdLicence
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.LicenceEvent
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.AuditEventRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceEventRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.ReleaseDateService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.AuditEventType
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceEventType
+import java.time.Clock
+import java.time.LocalDate
+
+@Service
+class TimeOutLicencesService(
+  private val licenceRepository: LicenceRepository,
+  private val releaseDateService: ReleaseDateService,
+  private val auditEventRepository: AuditEventRepository,
+  private val licenceEventRepository: LicenceEventRepository,
+  private val clock: Clock,
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  fun timeOutLicencesJob() {
+    log.info("Job to runTimeOutLicencesService started")
+    val jobExecutionDate = LocalDate.now(clock)
+    if (releaseDateService.excludeBankHolidaysAndWeekends(jobExecutionDate)) {
```

Comment:
mmh.. is this needed?

Reply:
Yes, If the job runs on bank holidays or weekends it will make the licences TIMED_OUT for next two working days which is not required.
Any way when the job runs on immediate working day after bank holiday or weekend then it will make the licences TIMED_OUT if eligible.